### PR TITLE
Add async standalone function example

### PR DESCRIPTION
**Pull request by GretaCB**
Per https://github.com/mapbox/node-cpp-skel/pull/44 and https://github.com/mapbox/node-cpp-skel/pull/38, this PR adds an _async_ standalone function example.

Note: It was super easy to integrate a new example into the current structure 🙌 

### Next
- [x] Add docs for this example
- [x] Add inline comments to give more context for using NAN's AsyncWorker

@daniel-j-h Mind taking a peek?

cc @springmeyer @mapsam

### CODE REVIEW

File: src/standalone_async/hello_async.cpp [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+namespace standalone_async {
+
+  /*
+  * This is an internal function used to return callback error messages instead of
+  * throwing errors.
+  * Usage:
+  *
+  * v8::Local<v8::Function> callback;
+  * CallbackError("error message", callback);
+  * return; // this is important to prevent duplicate callbacks from being fired!
```
Comment (daniel-j-h) [inline src/standalone_async/hello_async.cpp]: `return CallbackError("error message", callback)` works, too

File: src/standalone_async/hello_async.cpp [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      std::map<std::size_t,std::string> container;  
+
+      for (std::size_t i=0;i<work_to_do;++i) {
+          container.emplace(i,std::to_string(i));
+      }  
+
+      for (std::size_t i=0;i<work_to_do;++i) {
+          std::string const& item = container[i];
+          if (item != std::to_string(i)) {
+              abort(); // should never get here
```
Comment (daniel-j-h) [inline src/standalone_async/hello_async.cpp]: Hm this kills the v8 vm hard, no? Not sure if we should put this into the skeleton code for people to c&p.

File: src/standalone_async/hello_async.cpp [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+    bool louder = false;
+
+    // check second argument, should be a 'callback' function.
+    // This allows us to set the callback so we can use it to return errors
+    // instead of throwing as well.
+    if (!info[1]->IsFunction())
+    {
+        Nan::ThrowTypeError("second arg 'callback' must be a function");
+        return;
```
Comment (daniel-j-h) [inline src/standalone_async/hello_async.cpp]: We could `return Nan::ThrowTypeError("");` here and below, too.

File: src/standalone_async/hello_async.cpp [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+    // Create a worker instance and queues it to run asynchronously invoking the callback when done.
+    // - Nan::AsyncWorker takes a pointer to a Nan::Callback and deletes the pointer automatically.
+    // - Nan::AsyncQueueWorker takes a pointer to a Nan::AsyncWorker and deletes the pointer automatically.
+    auto* worker = new AsyncHelloWorker{louder, new Nan::Callback{callback}};
+    Nan::AsyncQueueWorker(worker);
+
+    // // "info" comes from the NAN_METHOD macro, which returns differently
+    // // according to the version of node
+    // info.GetReturnValue().Set(Nan::New<v8::String>("world").ToLocalChecked());
```
Comment (daniel-j-h) [inline src/standalone_async/hello_async.cpp]: I think this block can be remove - we want the function to return nothing and instead have the async worker call the user-provided callback

File: src/standalone_async/hello_async.cpp (hunk 2 above)
Comment (GretaCB) [inline src/standalone_async/hello_async.cpp]: @daniel-j-h great point. I [replaced it with a Callback](https://github.com/mapbox/node-cpp-skel/pull/46/commits/ec533587ef18d79647eb0fa280f4d372b4af923f#diff-881a9b2c4458d0d79b1d20ac96bba548R38), now troubleshooting compile errors.

Comment (GretaCB) [inline src/standalone_async/hello_async.cpp]: @daniel-j-h Nevermind, got it going!

**Comment by GretaCB** [comment]
Thanks for your comments @daniel-j-h !!! 🙌

**Pull request merged**